dashboard/ltiv1p3.py

```diff
@@ -0,0 +1,144 @@
+                      "membership/Instructor#TeachingAssistant" in role]
+    if len(teaching_roles) == 0:
+        logger.info(f"user {username} is not instructor in course {course_id}")
+        return False
+
+    ta = [teach_role for teach_role in teaching_roles if "membership/Instructor#TeachingAssistant" in teach_role]
+    if len(ta) != 0:
+        logger.info(f"user {username} is not a instructor of course {course_id}")
+        return False
+    else:
+        logger.info(f"user {username} is instructor in course {course_id}")
+        return True
+
+
+def validate_lti_1_3_request(request, message_launch):
+    launch_id = message_launch.get_launch_id()
+    launch_data = message_launch.get_launch_data()
+    custom_params = launch_data['https://purl.imsglobal.org/spec/lti/claim/custom']
+    course_name = launch_data['https://purl.imsglobal.org/spec/lti/claim/context']['title']
+    roles = launch_data['https://purl.imsglobal.org/spec/lti/claim/roles']
+    username = custom_params['user_username']
+    course_id = custom_params['canvas_course_id']
+    email = launch_data['email']
+    first_name = launch_data['given_name']
+    last_name = launch_data['family_name']
+    user_img = launch_data['picture']
+    RANDOM_PASSWORD_DEFAULT_LENGTH = 32
+
+    # Logging the user regardless course exits or not otherwise Django will show default login page
+    try:
+        user_obj = User.objects.get(username=username)
+    except User.DoesNotExist:
+        password = ''.join(random.sample(string.ascii_letters, RANDOM_PASSWORD_DEFAULT_LENGTH))
+        user_obj = User.objects.create_user(username=username, email=email, password=password, first_name=first_name, last_name=last_name)
+    user_obj.backend = 'django.contrib.auth.backends.ModelBackend'
+    django.contrib.auth.login(request, user_obj)
+
+    is_instructor = check_if_instructor(roles, username, course_id)
+
+    try:
+        course_details = Course.objects.get(canvas_id=course_id)
+    except ObjectDoesNotExist:
+        if is_instructor:
+            canvas_long_id = canvas_id_to_incremented_id(course_id)
+            course_details = Course.objects.create(id=canvas_long_id, canvas_id=course_id, name=course_name)
+            CourseViewOption.objects.create(course_id=canvas_long_id)
+
+    if course_details.term_id is None:
+        logger.info(f"Course {course_id} don't have a cron run yet")
+    lti_globals = {
+        "username": username,
+        "launch_id": launch_id,
+        "user_courses_info": {"course_id": course_id, "course_name": course_name},
+        "is_superuser": user_obj.is_superuser,
+        "login": settings.LOGIN_URL,
+        "logout": settings.LOGOUT_URL,
+        "user_image": user_img,
+        "primary_ui_color": settings.PRIMARY_UI_COLOR,
+        "help_url": settings.HELP_URL,
+        "google_analytics_id": settings.GA_ID
+    }
+    return lti_globals
+
+
+@csrf_exempt
+def login(request):
+    logger.info("LTI Login called...........")
```

Comment:
Not a very useful log message.  Try:

```suggestion
    logger.debug(f'request data: {request}')
```

The logging configuration should automatically include the module and method name in the log message.